homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -55,14 +56,14 @@ def unique_id(self):
     @property
     def state(self):
         """Return the state of the device."""
-        state = None
-        if self.type == "ping":
-            state = self.coordinator.data["ping"]
-        elif self.type == "download":
-            state = round(self.coordinator.data["download"] / 10 ** 6, 2)
-        elif self.type == "upload":
-            state = round(self.coordinator.data["upload"] / 10 ** 6, 2)
-        return state
+        if self.coordinator.data:
+            if self.type == "ping":
+                self._state = self.coordinator.data["ping"]
+            elif self.type == "download":
+                self._state = round(self.coordinator.data["download"] / 10 ** 6, 2)
+            elif self.type == "upload":
+                self._state = round(self.coordinator.data["upload"] / 10 ** 6, 2)
```

Comment:
This logic should move to `async_update`. We should not have side effects in a property.

Reply:
When the coordinator updates its data does it call `async_update`? I tried moving this to `async_update` but this is not called.

Reply:
You subscribe to the update coordinator, just subscribe a method that does this logic or remove writing to self._state